File: level1/K번째수/jihyung.py
def solution1(array, commands):
    answer = []
    for i in range(len(commands)):
        # start, end, 선택할 idx를 미리 정의함
        start = commands[i][0] - 1
        end = commands[i][1]
        idx = commands[i][2] - 1

        after = array[start:end]
        after.sort()
        answer.append(after[idx])

    return answer


def solution2(array, commands):
    answers = []
    for command in commands:
        # 따로 위치 정의 없이 인덱스로 접근한후에 나온 수로 그대로 슬라이싱
        sliced_array = array[command[0]-1:command[1]]
        sliced_array.sort()
        answers.append(sliced_array[command[2]-1])
    return answers

File: programmers/level1/12.K번째수/jihyung.java
import java.util.Arrays;
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        ArrayList<Integer> answer = new ArrayList<>();
        for(int[] command :commands){
            int start = command[0] -1;
            int end = command[1];
            int idx = command[2] -1;
            
           int[] sliced_array = Arrays.copyOfRange(array, start, end);
            Arrays.sort(sliced_array);
            answer.add(sliced_array[idx]);
        }
        int[] casted_answer = new int[answer.size()];
        for(int i=0; i<answer.size(); i++){
            casted_answer[i] = answer.get(i);
        }
        return casted_answer;
    }
}